### ISSUE TITLE
Performance guidelines - can we add them to the doc?

### ISSUE THREAD

**Issue by abergs** [s-panferov/awesome-typescript-loader#361]
Hi! I'd really like to gather some performance guidelines somewhere.

TL;DR: How do I squeeze performance out TS+webpack?

Currently we're running a webpack + react + typescript (tsx files) project and with 1200 modules it takes around 2.5 minutes to compile. Could we document better in what way useBabel affects the build-process, in what cases transpileOnly affects tsc. 

Is it possible to compile using all my cores? etc.

**Comment by krtr** [comment]
>Is it possible to compile using all my cores? etc.

potentially https://github.com/amireh/happypack but seems, that it doesn't work with this loader

**Comment by abergs** [comment]
@krtr Yep, been looking at happypack but as you said it currently do not work with atl. Or webpack2 for that matter.

**Comment by abergs** [comment]
We saw great improvements in compile time (1.5x) by doing:

```
"awesomeTypescriptLoaderOptions": {
        "forkChecker": true,
        "useTranspileModule": true,
        "useBabel":true,
        "useCache":true
    }
```

and we also added ?cacheDirectory=.awcache on our babel-loader, which might also increased the perf.
